email.go

```diff
@@ -188,27 +195,97 @@ func (em *Sender) buildMessage(text string, params Params) (message string, err
 	message = addHeader(message, "From", params.From)
 	message = addHeader(message, "To", strings.Join(params.To, ","))
 	message = addHeader(message, "Subject", params.Subject)
-	message = addHeader(message, "Content-Transfer-Encoding", "quoted-printable")
 
 	if em.contentType != "" {
 		message = addHeader(message, "MIME-version", "1.0")
-		message = addHeader(message, "Content-Type", fmt.Sprintf("%s; charset=%q", em.contentType, em.contentCharset))
 	}
+
 	message = addHeader(message, "Date", em.timeNow().Format(time.RFC1123Z))
 
 	buff := &bytes.Buffer{}
 	qp := quotedprintable.NewWriter(buff)
-	if _, err := qp.Write([]byte(text)); err != nil {
-		return "", err
+	mp := multipart.NewWriter(buff)
+	boundary := mp.Boundary()
+
+	withAttachments := len(params.Attachments) > 0
+	if withAttachments {
+		message = addHeader(message, "Content-Type", fmt.Sprintf("multipart/mixed; boundary=%q\r\n\r\n%s\r",
```

Comment:
unless I'm missing something, this is the second content type, in addition to the one below.  it doesn't look sane to me

Reply:
In case with attachments we set first content type to ```multipart/mixed``` with boundary value and set content type for each part next to boundary including body part. Like example bellow:

```
From: utkin.v.vitaliy@yandex.ru
To: "utkin.v.vitaliy@yandex.ru" <utkin.v.vitaliy@yandex.ru>
Subject: test with attachments
MIME-Version: 1.0
Date: Sat, 09 Apr 2022 13:28:31 +0500
Content-Type: multipart/mixed;
	boundary="----==--bound.93159.sas1-30406100349c.qloud-c.yandex.net"

------==--bound.93159.sas1-30406100349c.qloud-c.yandex.net
Content-Transfer-Encoding: 7bit
Content-Type: text/html

<div>test</div>
------==--bound.93159.sas1-30406100349c.qloud-c.yandex.net
Content-Disposition: attachment;
	filename="1.txt"
Content-Transfer-Encoding: base64
Content-Type: text/plain;
	name="1.txt"

dGVzdCBhdHRhY2htZW50Cg==
------==--bound.93159.sas1-30406100349c.qloud-c.yandex.net--
```

I can make ```addHeader``` helper package visible and set body headers in ```writeBody``` method for more readability.

Reply:
ok, makes sense

Reply:
Do you expect me to do small refactoring with ```addHeader``` helper?